RFS/src/main/java/com/rfs/common/http/GzipRequestTransformer.java

```diff
@@ -0,0 +1,75 @@
+package com.rfs.common.http;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.Deflater;
+import java.util.zip.GZIPOutputStream;
+
+import io.netty.handler.codec.http.HttpHeaderNames;
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import reactor.core.publisher.Mono;
+
+@AllArgsConstructor
+@Slf4j
+public class GzipRequestTransformer implements RequestTransformer {
```

Comment:
I'm not sure why we are using a separate transformer for this. `HttpClient.create()` supports `.compress(true/false)` that creates a client that uses compression, can we use that instead of building our own?

Reply:
That's to request/handle response compression, not request.

The main reason to compress is to reduce bandwidth used between RFS and Target. From the screenshot in the overview, requests can account for 75% of the total bandwidth so there is more opportunity to reduce bottlenecks with request compression.

There may be some value in enabling response compression by default too, it's less likely to have unintended consequences since the server can just ignore the request if it's not supported.

Reply:
Thanks for the details